### PR TITLE
feat(query-builder): Omit the clear action when empty

### PR DESCRIPTION
**Pull request by leeandher**
~Adds `hideClearWhenEmpty` as a prop to omit the action button to clear the query if nothing has been added to the query yet.~

Omits the action button when the query is empty.

### CODE REVIEW

File: static/app/components/searchQueryBuilder/index.spec.tsx [hunk 1]
```diff
@@ -221,6 +221,33 @@ describe('SearchQueryBuilder', function () {
...
+      const mockOnChange = jest.fn();
+      render(
+        <SearchQueryBuilder
+          {...defaultProps}
+          onChange={mockOnChange}
+          hideClearWhenEmpty
+        />
+      );
+      // Must await something to prevent act warnings
+      await act(tick);
```
Comment (scttcper) [inline static/app/components/searchQueryBuilder/index.spec.tsx:235]: just put a await findBy before other things
```suggestion
      await screen.findByRole('combobox', {name: 'Add a search term'});
```

File: static/app/components/searchQueryBuilder/index.spec.tsx [hunk 2]
```diff
@@ -221,6 +221,33 @@ describe('SearchQueryBuilder', function () {
...
+      ).not.toBeInTheDocument();
+      await userEvent.type(
+        screen.getByRole('combobox', {name: 'Add a search term'}),
+        'foo a:b{enter}'
+      );
+
+      await waitFor(() => expect(mockOnChange).toHaveBeenCalled());
+
+      expect(
+        screen.queryByRole('button', {name: 'Clear search query'})
```
Comment (scttcper) [inline static/app/components/searchQueryBuilder/index.spec.tsx:248]:
```suggestion
        screen.getByRole('button', {name: 'Clear search query'})
```

**Review by scttcper — APPROVED** [review]

File: static/app/components/searchQueryBuilder/index.tsx [hunk 3]
```diff
@@ -143,31 +147,32 @@ function SearchIndicator({
...
-          icon={<IconClose />}
-          borderless
-          onClick={() => {
-            dispatch({type: 'CLEAR'});
-            handleSearch('');
-          }}
-        />
-      </ButtonsWrapper>
-    );
+  if (disabled || (hideClearWhenEmpty && query === '')) {
```
Comment (malwilley) [inline static/app/components/searchQueryBuilder/index.tsx:156]: We still want to render `trailingItems` when the query is empty

**Review by malwilley — COMMENTED** [review]
Overall looks good, but I don't think we need this to be configurable. We could just enable this for all search bars, I don't see a downside.

File: static/app/components/searchQueryBuilder/index.tsx (hunk 3 above)
Comment (leeandher) [reply static/app/components/searchQueryBuilder/index.tsx:156]: Sounds good! i'll remove the prop and still render trailing items 👍

**Pull request merged**